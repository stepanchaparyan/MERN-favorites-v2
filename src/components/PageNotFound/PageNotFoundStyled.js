import styled from 'styled-components';
import { Link } from 'react-router-dom';
import PNF from '../../assets/404.jpg';

export const Container = styled.div`
  text-align: center;
  font-weight: 600;
  font-size: 24px;
  background-image: url(${PNF});
  background-position: center;
  background-repeat: no-repeat;
  background-size: contain;
  height: 70vh;
`;

export const Module = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  padding: 16px;
`;

export const LinkStyled = styled(Link)`
  color: ${props => props.theme.cadetblue};
  margin: 32px;
  text-decoration: none;
  font-size: 28px;
  position: absolute;
  bottom: 10vh;
`;
