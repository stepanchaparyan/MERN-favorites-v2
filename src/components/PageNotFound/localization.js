import { defineMessages } from 'react-intl';

export default defineMessages({
  goHomePage: {
    id: 'pageNotFound.goHomePage',
    defaultMessage: 'Go Home page'
  }
});
