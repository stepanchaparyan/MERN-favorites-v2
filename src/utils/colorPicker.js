import theme from '../styles/theme';
import { FAVITEM } from '../constants';

const { red, blue, limeGreen, orange, black } = theme;
const { TYPE } = FAVITEM;

export const colorPicker = category => {
  let color;
  switch (category) {
    case TYPE.FILM:
      color = red;
      break;
    case TYPE.MUSIC:
      color = blue;
      break;
    case TYPE.BOOKS:
      color = limeGreen;
      break;
    case TYPE.OTHER:
      color = orange;
      break;
    default:
      color = black;
  }
  return color;
};
