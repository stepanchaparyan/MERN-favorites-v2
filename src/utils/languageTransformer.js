const dict = {
  // lang - country
  ar: 'sa',
  be: 'bl',
  cs: 'sz',
  da: 'dk',
  de: 'de',
  el: 'gr',
  en: 'us',
  es: 'es',
  fa: 'ir',
  fr: 'fr',
  ja: 'jp',
  hi: 'in',
  hu: 'hu',
  hr: 'hr',
  hy: 'am',
  it: 'it',
  kk: 'kz',
  nl: 'nl',
  no: 'no',
  pl: 'pl',
  pt: 'pt',
  ru: 'ru',
  sk: 'sk',
  sl: 'sl',
  sv: 'se',
  tr: 'tr',
  uk: 'ua'
};

export const languageTransformer = listList => {
  let newList = [];
  listList.map(lang => {
    const item = Object.keys(dict).find(item => item === lang);
    if (item) {
      newList.push(dict[item].toUpperCase());
    }
  });
  return newList;
};

function getKeyByValue(object, value) {
  return Object.keys(object).find(key => object[key] === value);
}

export const setLanguage = (code, setLocale) => {
  const countryCode = Object.values(dict).includes(code.toLowerCase());

  if (countryCode) {
    setLocale(getKeyByValue(dict, code.toLowerCase()));
  }
};
