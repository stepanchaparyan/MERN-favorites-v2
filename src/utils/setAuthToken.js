import axios from 'axios';

const Authorization = 'Authorization';

const setAuthToken = token => {
  if (token) {
    axios.defaults.headers.common = { Authorization: `Bearer ${token}` };
  } else {
    delete axios.defaults.headers.common[Authorization];
  }
};
export default setAuthToken;
