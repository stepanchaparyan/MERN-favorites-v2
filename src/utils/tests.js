import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import { shallow } from 'enzyme';

export function renderToJSON(Component, props = {}, Router) {
  if (Router) {
    return renderer
      .create(
        <Router>
          <Component {...props} />
        </Router>
      )
      .toJSON();
  }
  return renderer.create(<Component {...props} />).toJSON();
}

export function renderSnapshot(Component, props = {}) {
  const wrapper = shallow(<Component {...props} />);
  expect(wrapper).toMatchSnapshot();
}
