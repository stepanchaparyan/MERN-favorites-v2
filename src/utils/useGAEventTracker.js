// eslint-disable-next-line no-unused-vars
import React from 'react';
import ReactGA from 'react-ga';

const useGAEventTracker = (category = 'Category') => {
  return (action = 'action', label = 'label') => {
    ReactGA.event({
      category,
      action,
      label,
    });
  };
};

export default useGAEventTracker;
