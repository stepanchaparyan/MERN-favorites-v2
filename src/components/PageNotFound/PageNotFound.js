import React from 'react';
import { useIntl } from 'react-intl';
import { Container, Module, PageNotFoundText, LinkStyled, Logo } from './PageNotFoundStyled';
import Img from '../../assets/elephant.png';
import localization from './localization';
import { LINK } from '../../constants';

const NotFound = () => {
  const { formatMessage } = useIntl();

  return (
    <Container>
      <Module>
        <PageNotFoundText>{formatMessage(localization.pageNotFound)}</PageNotFoundText>
        <LinkStyled to={LINK.TO.WELCOME}>{formatMessage(localization.goHomePage)}</LinkStyled>
        <Logo src={Img}></Logo>
      </Module>
    </Container>
  );
};
export default NotFound;
