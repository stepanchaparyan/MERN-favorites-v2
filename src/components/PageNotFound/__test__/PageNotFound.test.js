import { renderSnapshot } from '../../../utils/tests';
import PageNotFound from '../PageNotFound.js';

jest.mock('react-intl', () => ({
  useIntl: jest.fn(() => ({ formatMessage: m => m.defaultMessage })),
  defineMessages: i => i
}));

describe('PageNotFound component test with Enzyme', () => {
  test('renders snapshot', () => {
    renderSnapshot(PageNotFound);
  });
});
