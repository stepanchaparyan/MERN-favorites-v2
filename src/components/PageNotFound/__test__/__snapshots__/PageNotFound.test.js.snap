// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound component test with Enzyme renders snapshot 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      Page not found
    </styled.div>
    <Styled(Link)
      to="/"
    >
      Go Home page
    </Styled(Link)>
    <styled.img
      src="test-file-stub"
    />
  </styled.div>
</styled.div>
`;
